sdk/storage/Azure.Storage.Queues/src/QueueClient.cs

```diff
@@ -221,6 +251,28 @@ public QueueClient(string connectionString, string queueName, QueueClientOptions
             _clientSideEncryption = QueueClientSideEncryptionOptions.CloneFrom(options._clientSideEncryptionOptions);
             _storageSharedKeyCredential = conn.Credentials as StorageSharedKeyCredential;
             _messageEncoding = options.MessageEncoding;
+            _name = uriBuilder.QueueName;
+
+            uriBuilder.QueueName = null;
+
+            _queueRestClient = new QueueRestClient(
+                _clientDiagnostics,
+                _pipeline,
+                url: uriBuilder.ToUri().ToString(),
+                _name,
+                _version.ToVersionString());
```

Comment:
I'd create local variables for these, we're calling these repeatedly. (applies to other places like that).

This tripple client init looks like something that could be extracted to private method and reused in these constructors.

Reply:
fixed.